### ISSUE TITLE
Investigate profile guided optimisation

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#331]
https://blog.rust-lang.org/inside-rust/2020/11/11/exploring-pgo-for-the-rust-compiler.html

**Comment by Firstyear** [comment]
https://doc.rust-lang.org/stable/rustc/profile-guided-optimization.html

**Comment by sn99** [comment]
Hi, is this open for GSoC ?

**Comment by Firstyear** [comment]
Yes - please look at the project offering again and contact me about it.

**Comment by sn99** [comment]
Thanks for the reply, I have sent a mail with the link to my GSoC draft
